### DISCUSSION TITLE
METviewer plotting of adding users defined line and marker (symbol)

### DISCUSSION THREAD

**Discussion by Ho-ChunHuang-NOAA** [dtcenter/METplus#1563] [Plot Generation] [answered]
Hi,

I would like to know whether the MetViewer python code support users  adding a line such as zero line in ME plot or a symbol/marker to any location of the figure.  The older command for R-script seems not working.  Is there a replacement command?

<plot_cmd>points(1,1,pch=19,col=\"magenta\",cex=2)</plot_cmd>

<plot_cmd>abline(h=0, col=\"black\", lwd=2, lty=2)</plot_cmd>

**Comment by TatianaBurek** [comment, marked as answer]
Yes, it is possible to add horizontal and vertical lines to the plots. Adding symbol/marker is not implemented.
To add a line using UI:
- click on "+ Add line" on the bottom of the page
- in the pop-up select the line type (horizontal or vertical) and the position of the line - 0 for the zero line
- in the bottom table choose the line color, width, type
![Screen Shot 2022-04-08 at 07 58 13](https://user-images.githubusercontent.com/21346768/162452289-dcb4b316-7e4a-4ec4-80bf-c611a134adeb.png)

To add a line using XML:
- add an element <lines> to the <plot> section. Example:
 ```
<lines>
            <line color="#ff00ff" line_pos="0" lty="1" lwd="1" type="horiz_line"/>
 </lines>
```
The example plot - #20220408_135731

**Comment by Ho-ChunHuang-NOAA** [comment]
Hi, Tatiana:

Thanks for the add line option.  I am able to add zero line onto ME plots.

Ho-Chun Huang, Ph.D.

IMSG at NOAA/NWS/NCEP/EMC

5830 University Research Ct., Rm. 2792

College Park, MD 20740

***@***.*** ***@***.***>

301-683-3958
